### PR TITLE
Next.js: Enable SWC for new Webpack5-based projects

### LINKED ISSUE #24869: Adjust sb init to set useSWC to true for all new Webpack5-based projects, except for Angular

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/storybook/issues/24869

## What I did

`sb init` will set `true` for the `useSWC` flag for all new Webpack5-based projects, except for Angular and Next.js

#### The changes in this PR are covered in the following automated tests:
- [x] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

1. Run a Webpack5 sandbox
2. Go to `.storybook/main.<ts|js>` and check, whether `framework: { builder: { useSWC: true }}` is set  (except for Angular and Next.js)

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: code/lib/cli/src/generators/ANGULAR/index.ts [hunk 1]
```diff
@@ -13,10 +12,6 @@ const generator: Generator<{ projectName: string }> = async (
   options,
   commandOptions
 ) => {
-  const angularVersion = await packageManager.getPackageVersion('@angular/core');
-  const isWebpack5 = angularVersion && semver.gte(angularVersion, '12.0.0');
```
Comment (valentinpalkovic) [inline code/lib/cli/src/generators/ANGULAR/index.ts:17]: We don't support versions lower than Angular 15.

File: code/lib/cli/src/generators/baseGenerator.ts [hunk 2]
```diff
@@ -171,6 +171,17 @@ const hasInteractiveStories = (rendererId: SupportedRenderers) =>
 const hasFrameworkTemplates = (framework?: SupportedFrameworks) =>
   ['angular', 'nextjs'].includes(framework);
 
+function shouldUseSWCCompiler(builder: Builder, projectType: ProjectType) {
+  console.log({ builder, projectType });
```
Comment (yannbf) [inline code/lib/cli/src/generators/baseGenerator.ts:175]:
```suggestion
```

**Review by yannbf — APPROVED** [review]

File: code/frameworks/nextjs/src/swc/next-swc-loader-patch.ts [hunk 3]
```diff
@@ -165,11 +165,20 @@ export function pitch(this: any) {
   }, callback);
 }
 
+function sanitizeSourceMap(rawSourceMap: any): any {
+  const { sourcesContent, ...sourceMap } = rawSourceMap ?? {};
+
+  // JSON parse/stringify trick required for swc to accept the SourceMap
+  return JSON.parse(JSON.stringify(sourceMap));
+}
```
Comment (ndelangen) [inline code/frameworks/nextjs/src/swc/next-swc-loader-patch.ts:173]: Is this an issue somewhere?

File: code/lib/cli/src/generators/baseGenerator.ts [hunk 4]
```diff
@@ -171,6 +171,15 @@ const hasInteractiveStories = (rendererId: SupportedRenderers) =>
...
   ['angular', 'nextjs'].includes(framework);
 
+function shouldUseSWCCompiler(builder: Builder, projectType: ProjectType) {
+  return (
+    builder === CoreBuilder.Webpack5 &&
+    projectType !== ProjectType.ANGULAR &&
+    // TODO: Remove in Storybook 8.0
+    projectType !== ProjectType.NEXTJS
+  );
+}
```
Comment (ndelangen) [inline code/lib/cli/src/generators/baseGenerator.ts:181]: I wonder if this should be a setting in each individual generator?

Like here:
https://github.com/storybookjs/storybook/blob/52fab185a9e30b6c69758b47264f1b56a9bd9ef8/code/lib/cli/src/generators/HTML/index.ts#L5

**Pull request merged**